src/oracle_implementations/discount_rate/Yield/YieldValueProvider.sol

```diff
@@ -66,6 +72,13 @@ contract YieldValueProvider is Oracle, Convert {
         // Compute the elapsed time
         uint32 timeElapsed = blockTimestamp - blockTimestampLast;
 
+        // ensure that at least one full period has passed since the last update
+        if (timeElapsed >= period) {
```

Comment:
The update window is taken into account in the oracle base implementation. So getValue will not get called inside the same updateWindow. I don't think this check is necessary. 
Here is the base oracle implementation: https://github.com/fiatdao/delphi/blob/58047725f84fca5f2f61ddf66beb4a7aa99c0aa7/src/oracle/Oracle.sol#L88